projects/hr-portal/src/hr-portal.ts

```diff
@@ -1,83 +1,195 @@
 import { LitElement, css, html } from "lit";
 import { customElement, property } from "lit/decorators.js";
+import "igniteui-webcomponents-grids/grids/combined.js";
+import { configureTheme, defineComponents, IgcAvatarComponent, IgcIconButtonComponent, IgcIconComponent, registerIcon } from "igniteui-webcomponents";
+import { IgcTreeGridComponent, IgcCellTemplateContext } from "igniteui-webcomponents-grids/grids";
+import { icons } from "../public/images/Icons";
+
+
+defineComponents(IgcAvatarComponent, IgcIconComponent, IgcTreeGridComponent, IgcIconButtonComponent);
+configureTheme("fluent");
 
 @customElement("app-hr-portal")
 export default class HrPortal extends LitElement {
-  render() {
-    return html` <div>HR PORTAL</div> `;
+
+  @property({ type: Array })
+  data: Array<any> = [];
+
+  constructor() {
+    super();
+    this.data = [];
+    icons.forEach((icon: { name: string; path: string; category: string }) => {
+      registerIcon(icon.name, icon.path, icon.category);
+    });
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.fetchData();
+  }
+
+  async fetchData() {
+    try {
+      const response = await fetch('https://staging7.infragistics.com/grid-examples-data/data/hr/hr.json');
+      const jsonData = await response.json();
+      console.log('Fetched data:', jsonData); // Log the fetched data
+      this.data = jsonData; 
+      console.log('Assigned data:', this.data); // Log the assigned data
+    } catch (error) {
+      console.error('Error fetching HR data:', error);
+    }
+  }
+
+  clearSorting() {
+    const treeGrid = this.shadowRoot?.getElementById('treeGrid') as IgcTreeGridComponent;
+    if (treeGrid) {
+      treeGrid.sortingExpressions = [];
+    }
+  }
+  
+  public avatarTemplate = (ctx: IgcCellTemplateContext) => {	
+    let row = ctx.cell.row;
```

Comment:
isnt this the same as `ctx.cell.value`? why get the value from the row. Same as all other body templates

Reply:
no, the value is the name, the avatar is looking for picture path which is another property. Same goes for other templates. The contact template takes value from 3 fields.